iphone/Classes/TiUIListView.m

```diff
@@ -2023,9 +2027,7 @@ - (void)searchBarTextDidEndEditing:(UISearchBar *)searchBar
   if ([searchBar.text length] == 0) {
     self.searchString = @"";
     [self buildResultsForSearchText];
-    if ([searchController isActive]) {
-      [searchController setActive:NO];
-    }
+    [self performSelector:@selector(dismissSearchController) withObject:nil afterDelay:.2];
```

Comment:
Why do we need a delay here?

Reply:
Would it make sense to introduce a flag to check for this case? Using a delay may be error prone when the transition takes longer than expected for whatever reason or Apple decides to change the transition duration.

As far as i understand this would also introduce an unnecessary delay if the user just empties the search string and ends editing, wouldn't it?

Reply:
I don't think we should introduce a new flag. We are checking [searchController isActive] for active state. This is an edge case in which this issue is raised. Actually there needs only time difference to get value updated. When we end editing, Search controller delegate - (void)willDismissSearchController:(UISearchController *)searchController  first called before
- (void)searchBarTextDidEndEditing:(UISearchBar *)searchBar. 
So it may be possible that there is no need to call
searchController.active = NO; 

ListView's Search functionality needs refactoring, I'll plan in future.

Reply:
Alright, that sounds reasonable. Let's take the fix as is and you'll schedule a refactor.